lib/associations/has-many.js

```diff
@@ -84,6 +84,11 @@ class HasMany extends Association {
     if (this.source.rawAttributes[this.sourceKey]) {
       this.sourceKeyAttribute = this.sourceKey;
       this.sourceKeyField = this.source.rawAttributes[this.sourceKey].field || this.sourceKey;
+      if (this.sequelize.options.dialect === 'db2' &&
```

Comment:
This should not be required, other dialects have this limitation as well (you can't reference non-unique column as a foreign key).

Reply:
@sushantdhiman Addressed it in latest [commit](https://github.com/ibmdb/sequelize/commit/df7fae1086f11bd75e1e94e77c05799cb0f9542a). Removed check for db2.  The test code of [has-many.test.js](https://github.com/sequelize/sequelize/blob/master/test/integration/associations/has-many.test.js#L1510) has not specified email as non null and unique, so need to take care by the code.

Reply:
`indexes` should define unique constrain on `email`, https://github.com/sequelize/sequelize/blob/2ae60889d6b1f46585fdc3b82e9927edef541f8f/test/integration/associations/has-many.test.js#L1511

Perhaps db2 implementation is not taking care of `indexes` as part of `Model.sync`

Reply:
@sushantdhiman Addressed the issue in latest commit. Thanks.

Reply:
@sushantdhiman Please provide little bit more clarity about this comment as removing the code is causing few test failures. The Db2 implementation is taking care of indexes and for the mentioned test case, we can see create index statement is getting executed. In Db2, to create unique index, the column need not to be NOT NULL UNIQUE, but when we reference a column as foreign key, it must be NOT NULL UNIQUE. Below SQL works fine and it is able to create unique index for the mentioned test case:
```
CREATE TABLE "UserXYZs" ("id" INTEGER NOT NULL GENERATED BY DEFAULT AS IDENTITY(START WITH 1, INCREMENT BY 1) , "username" VARCHAR(255), "email" VARCHAR(255), "createdAt" TIMESTAMP NOT NULL, "updatedAt" TIMESTAMP NOT NULL, PRIMARY KEY ("id"));
CREATE UNIQUE INDEX "user_x_y_zs_email" ON "UserXYZs" ("email");
```
But below query fails with error: SQL0573N  A column list specified in the references clause of constraint "user_email_xyz..." does not identify a unique constraint of the parent table or nickname "NEWTON.UserXYZs".  SQLSTATE=42890
```
CREATE TABLE "TaskXYZs" ("id" INTEGER NOT NULL GENERATED BY DEFAULT AS IDENTITY(START WITH 1, INCREMENT BY 1) , "title" VARCHAR(255), "user_email_xyz" VARCHAR(255), "createdAt" TIMESTAMP NOT NULL, "updatedAt" TIMESTAMP NOT NULL,PRIMARY KEY ("id"), FOREIGN KEY ("user_email_xyz") REFERENCES "UserXYZs" ("email") ON DELETE CASCADE);
```
Above error says, when a column is referenced as foreign key, it must be unique, but to create a unique index, column need not be unique. So, in case of foreign key only when a column is mentioned as sourceKey, we are making it unique and hence added code in has-many.js file.
Do you mean, if a column is mentioned in the fields of unique index, the column must be UNIQUE irrespective of we mention that column as a foreign key or not?

Reply:
Btw, below SQL works fine because Db2 automatically create indexes if a column is defined as NOT NULL UNIQUE:
```
CREATE TABLE "UserXYZs" ("id" INTEGER NOT NULL GENERATED BY DEFAULT AS IDENTITY(START WITH 1, INCREMENT BY 1) , "username" VARCHAR(255), "email" VARCHAR(255) NOT NULL UNIQUE, "createdAt" TIMESTAMP NOT NULL, "updatedAt" TIMESTAMP NOT NULL, PRIMARY KEY ("id"));
CREATE TABLE "TaskXYZs" ("id" INTEGER NOT NULL GENERATED BY DEFAULT AS IDENTITY(START WITH 1, INCREMENT BY 1) , "title" VARCHAR(255), "user_email_xyz" VARCHAR(255), "createdAt" TIMESTAMP NOT NULL, "updatedAt" TIMESTAMP NOT NULL,PRIMARY KEY ("id"), FOREIGN KEY ("user_email_xyz") REFERENCES "UserXYZs" ("email") ON DELETE CASCADE);
```

Reply:
@sushantdhiman If you say that if column is mentioned under fields of unique indexes, such columns must be UNIQUE, then why `model\sync.test.js` is trying to insert duplicate values for the column `age` here: https://github.com/sequelize/sequelize/blob/master/test/integration/model/sync.test.js#L120 ?
Db2 do not allow duplicate values for unique columns.

Reply:
Kept this code in latest commit has this is the only place where we can inform source table that its column is used as a foreign key and that column should be unique. Otherwise, please read previous comment and suggest. Thanks.